### PR TITLE
[FABN-1647] Add mising type def

### PR DESCRIPTION
**Pull request by lesleyannjordan**
Signed-off-by: lesleyannj <lesleyannj@hotmail.com>

### CODE REVIEW

**Comment by lesleyannjordan** [comment]
FABN-1647

File: fabric-client/types/index.d.ts [hunk 1]
```diff
@@ -379,6 +379,7 @@ declare namespace Client { // tslint:disable-line:no-namespace
 		public lastBlockNumber(): number;
 		public isconnected(): boolean;
 		public connect(options?: ConnectOptions | boolean, connectCallback?: (err: Error, channelEventHub: ChannelEventHub) => void): void;
+		public reconnect(options?: ConnectOptions, connectCallback?: (err: Error, channelEventHub: ChannelEventHub) => void): void;
```
Comment (bestbeforetoday) [inline fabric-client/types/index.d.ts:382]: This calls through to `connect` so the options should be the same, i.e. `options?: ConnectOptions | boolean`. Unfortunately the TypeScript definition of `ConnectOptions` is also incomplete. See here for details of all the properties it should include: https://hyperledger.github.io/fabric-sdk-node/release-1.4/global.html#ConnectOptions

**Review by bestbeforetoday — CHANGES_REQUESTED** [review]

File: fabric-client/types/index.d.ts (hunk 1 above)
Comment (lesleyannjordan) [reply fabric-client/types/index.d.ts:382]: The reconnect function add a boolean set to true into the options.  So it what it passes onto connect is the same.  The comment for the reconnect function doesn't indicate can take a boolean in.

Comment (bestbeforetoday) [reply fabric-client/types/index.d.ts:382]: You're right, my mistake. `options?: ConnectOptions` is correct, not `options?: ConnectOptions | boolean` as I said above

Comment (lesleyannjordan) [reply fabric-client/types/index.d.ts:386]: Additions added for ConnectOptions

**Review by bestbeforetoday — APPROVED** [review]

**Pull request merged**